### DISCUSSION TITLE
How to deserialize a multipart/form-data response to a FormData instance?

### DISCUSSION THREAD

**Discussion by fgsalomon** [axios/axios#6180] [Q&A]
Hi,

I'd like to use axios in the browser to consume an EP of a service that returns a `multipart/form-data` response with a file and a JSON object.

I didn't find any example on the docs and I couldn't make it work. My current attempt is this:

```javascript
 sendRequest(file) {
    const form = new FormData();
    form.append("file", file);
    return axios.post("https://example.com", form, {
      headers: {"Accept": "multipart/form-data"}
    });
  }
```
but the Promise contains a string with the raw response instead of an instance of `FormData`.

Using `fetch` I'm able to return a `FormData` but I'd like to use axios instead since all my codebase depends on axios:
```javascript
async  sendRequest(file) {
    const form = new FormData();
    form.append("file", file);
     const response = await fetch("https://example.com",  {
        method: "POST",
        body: form
      });
	 return await response.formData();
  }
```

Thanks

**Comment by DigitalBrainJS** [comment]
Axios doesn't support formData response, but you can try to handle it manually (not tested):
```js
  const {data} = await axios.get(url, {responseType: 'arraybuffer'});
  const formdata = await new Response(data).formData();
```

**Reply by fgsalomon** [reply]
It doesn't seem to work. The response object is created, but retrieving the formData throws an exception `TypeError: Failed to fetch...`.

```
  const data = await axios.get(url, {responseType: 'arraybuffer'});
  const response = new Response(data);
  const formdata = await response.formData(); // This is where the exception is thrown
```

**Reply by inkognitro** [reply]
Thanks to @DigitalBrainJS, this actually worked for me by adding the according `content-type` header to the native Response object headers. The "boundary" definition is required to be able to parse a form data string. See also: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type#syntax)

```typescript
const response = await axios.get(url, {responseType: 'arraybuffer'});
const plainHeaders = response.headers instanceof AxiosHeaders
   ? response.headers.toJSON(true)
   : response.headers;
const nativeResponse = new Response(response.data, {
   headers: {
      'content-type': `${plainHeaders['content-type']}`,
   },
   status: response.status,
   statusText: response.statusText,
});
const formData = nativeResponse.formData();
```
Might be better to add all headers instead of just `content-type`.

**Reply by DigitalBrainJS** [reply]
@inkognitro You're right- the boundary header is required. In addition, since v1.7.0 you can use the fetch adapter with FormData response type.
```js
const response = await axios.get(url, {adapter: 'fetch', responseType: 'formdata'});
```
